summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -175,6 +185,10 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
         case OptionMappedProducer(_, op) =>
           val boxed = Externalizer(op)
           spout.flatMap { case (time, t) => boxed.get(t).map { x => (time, x) } }
+        case FlatMappedProducer(_, op) => {
+          val lockedOp = Externalizer(op)
+          spout.flatMap { case (time, t) => lockedOp.get.apply(t).map { x => (time, x) } }
```

Comment:
Let's use `boxed` and `boxed.get(t)` as above.  This isn't really performing any locking.